src/eko/matching_conditions/as2.py

```diff
@@ -157,14 +157,14 @@ def A_hg(n, sx, L):
         A_hg : complex
             |NNLO| heavy-gluon |OME| :math:`A_{Hg}^{S,(2)}`
     """
-    S1 = sx[0, 0]
-    S2 = sx[1, 0]
-    S3 = sx[2, 0]
+    S1 = sx[0][0]
+    S2, Sm2 = sx[1]
+    if len(sx[2]) == 3:
```

Comment:
This sounds a bit dangerous ... can't we unify the structure? Where does the branching come from?

Reply:
It's due to the different number of harmonics needed at NNLO and N3LO

Reply:
I see ...